doc/man3/SSL_CTX_new.pod

```diff
@@ -81,41 +81,51 @@ an existing B<SSL_CTX> structure.
 =head1 NOTES
 
 The SSL_CTX object uses B<method> as connection method.
+A list of available SSL/TLS methods is shown below.
+
 The methods exist in a generic type (for client and server use), a server only
 type, and a client only type.
-B<method> can be of the following types:
+
+The methods are further divided into I<general-purpose> methods and
+I<version-specific> methods: while the general-purpose methods will automatically
+negotiate the highest version mutually supported by the client and the server,
+the version-specific methods only support a single protocol version and will
+fail if that version is not supported by the peer.
+
+A significant disadvantage of using the version-specific methods is that
+an application will not automatically upgrade to newer protocol versions as
+they are added to OpenSSL, without having to be modified and recompiled.
+For that reason, applications should use only the general-purpose methods.
+In fact, all version-specific methods have been deprecated.
```

Comment:
I disagree with moving this text up from the items they originally came from, and I also think that if you really want to move this text around, you should do so in a different PR, where the pros and cons can be discussed without getting in the way of what you obviously want to do with this PR, i.e. mark the version specific methods as deprecated.

Reply:
Sigh... it looks like you are right and I attempted too much at once. The reason why I moved this part to the beginning was because I wanted to motivate why the version-specific functions were deprecated, and I wanted to do it only once in order not to repeat myself. But I think you are right, this change is wrong for two reasons:

- People will scroll (resp. search) forward straightaway to the function they intend to use and will be confused not to find the desired information right there, but a few lines above.
- People don't want to learn about the reasons why things were deprecated (at least not here in the manual page), they just want to learn quickly which functions they can use and how to use them.

So I'll revert most of this and only leave the deprecation notes.